packages/ts/react-form/test/useForm.spec.tsx

```diff
@@ -370,6 +371,19 @@ describe('@vaadin/hilla-react-form', () => {
       const count = await findByTestId('count');
       expect(count).to.have.text('1');
     });
+
+    it('should work with optional hierarchy with optional values', async () => {
+      function StreetAddressView() {
+        const { model, field } = useForm(PostalAddressModel);
+        useFormPart(model.streetAddress.name);
+
+        return <input data-testid="street-name" {...field(model.streetAddress.name)} />;
+      }
+
+      const { container } = render(<StreetAddressView />);
+      const streetNameField = await waitFor(() => container.querySelector('input')!);
+      expect(streetNameField.value).to.have.rendered.eq('undefined');
```

Comment:
I think this line could be simply changed to:
```suggestion
      expect(streetNameField.value).to.be.equal('undefined');
```
since this is actually the current behaviour when binding to a plain `<input />` element.
Related issue: #2658

Reply:
For making sure it remains compliant with the React Components, we can also render another component that renders a `<TextField />` and check that the `'undefined'` is taken care of in our library, by expecting the value of the `<TextField />` element is equal to `''`, not `'undefined'`.

Reply:
`TextField` test would add "react-components" dependency that would need have its own version as it is done in react-crud. 
I think the best action would be to fix https://github.com/vaadin/hilla/issues/2658

Reply:
OK, let's keep it as it is with `<input />`. I believe the `to.have.rendered.eq('undefined');` could still be simplified to `to.be.equal('undefined');`.